src/model-helpers/Document.js

```diff
@@ -0,0 +1,206 @@
+  const signingKey = await getUserSigningKey(session);
+  const signatureDataset = await signDocumentTtlFile(signingKey, document, session, containerUrl);
+  await saveSolidDatasetAt(`${containerUrl}signature.ttl`, signatureDataset, {
+    fetch: session.fetch
+  });
+};
+
+/**
+ * Helper Function that returns Driver's License ttl file based off of image passed
+ *
+ * @function createDriversLicenseTtlFile
+ * @memberof utils
+ * @function createDriversLicenseTtlFile
+ * @param {fileObjectType} fileObject - Object containing information about file
+ * @returns {Promise<ThingLocal>} TTL file Thing - Processes a barcode using zxing
+ * and returns a new TTL file Thing
+ */
+const createDriversLicenseTtlFile = async (fileObject) => {
+  const dlData = await getDriversLicenseData(fileObject.file);
+  return buildThing(createThing({ name: 'license information' }))
+    .addStringNoLocale(RDF_PREDICATES.additionalType, dlData.DCA)
+    .addStringNoLocale(RDF_PREDICATES.conditionsOfAccess, dlData.DCB)
+    .addDate(RDF_PREDICATES.expires, new Date(`${formattedDate(dlData.DBA)}`))
+    .addStringNoLocale(RDF_PREDICATES.givenName, dlData.DCS)
+    .addStringNoLocale(RDF_PREDICATES.alternateName, dlData.DAC)
+    .addStringNoLocale(RDF_PREDICATES.familyName, dlData.DAD)
+    .addDate(RDF_PREDICATES.dateIssued, new Date(`${formattedDate(dlData.DBD)}`))
+    .addDate(RDF_PREDICATES.dateOfBirth, new Date(`${formattedDate(dlData.DBB)}`))
+    .addStringNoLocale(RDF_PREDICATES.gender, dlData.DBC)
+    .addStringNoLocale(RDF_PREDICATES.Eye, dlData.DAY)
+    .addInteger(RDF_PREDICATES.height, Number(dlData.DAU))
+    .addStringNoLocale(RDF_PREDICATES.streetAddress, dlData.DAG)
+    .addStringNoLocale(RDF_PREDICATES.City, dlData.DAI)
+    .addStringNoLocale(RDF_PREDICATES.State, dlData.DAJ)
+    .addStringNoLocale(RDF_PREDICATES.postalCode, dlData.DAK)
+    .addStringNoLocale(RDF_PREDICATES.identifier, dlData.DAQ)
+    .addStringNoLocale(RDF_PREDICATES.identifier, dlData.DCF)
+    .addStringNoLocale(RDF_PREDICATES.Country, dlData.DCG)
+    .addStringNoLocale(RDF_PREDICATES.additionalName, dlData.DDE)
+    .addStringNoLocale(RDF_PREDICATES.additionalName, dlData.DDF)
+    .addStringNoLocale(RDF_PREDICATES.additionalName, dlData.DDG)
+    .build();
+};
+
+/**
+ * Function that generates checksum for uploaded file
+ *
+ * @memberof utils
+ * @function createFileChecksum
+ * @param {fileObjectType} file - Object containing information about file
+ * from form submission (see {@link fileObjectType})
+ * @returns {Promise<WordArray>} Promise - Generates checksum for uploaded file
+ * using the SHA256 algorithm
+ */
+const createFileChecksum = async (file) => {
+  const text = await file.text(); // only hash the first megabyte
+  return sha256(text);
+};
+
+const addAdditionalInfo = (thing, fileObject) => {
+  if (fileObject.type === "Driver's License") {
+    return thing.addUrl(RDF_PREDICATES.about, createDriversLicenseTtlFile(fileObject));
```

Comment:
Following the errors with Driver's License, this line might also be the issue for failing to upload a new container. The data is coming out properly, but .addUrl was expecting a URL rather than a Thing indicated by the [[object Promise]] in the clip.

https://github.com/codeforpdx/PASS/assets/14917816/8419b3e3-1bdd-4665-b506-0a9f0348ed98

Think we could solve this by chaining the two Things with setThing into one TTL file rather than putting inside the Thing for docDesc.name.

So we can build the Thing for docDesc.name first in `docDescToThing` and build the Thing with "license information" with `createDriversLicenseTtlFile(file)` inside `saveDescription` if `fileObject.type === "Driver's License"`, then set both Things with setThing to newDataset

Reply:
Fixed